src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -513,6 +525,71 @@ def getThresholds(self):
+            taskRunningSlots = self.currentRcThresholds[siteName]['thresholds'][jobType]["max_slots"]
+            taskRunningJobs = self.currentRcThresholds[siteName]['thresholds'][jobType]["task_running_jobs"]
+
+            # set the initial totalPendingJobs since it increases in every cycle when a job is submitted
+            self.currentRcThresholds[siteName].setdefault("init_total_pending_jobs", totalPendingJobs)
+
+            # set the initial taskPendingJobs since it increases in every cycle when a job is submitted
+            self.currentRcThresholds[siteName]['thresholds'][jobType].setdefault("init_task_pending_jobs", taskPendingJobs)
+
+            initTotalPending = self.currentRcThresholds[siteName]["init_total_pending_jobs"]
+            initTaskPending = self.currentRcThresholds[siteName]['thresholds'][jobType]["init_task_pending_jobs"]
+
+        except KeyError as ex:
+            msg = "Invalid key for site %s and job type %s\n" % (siteName, jobType)
+            logging.exception(msg)
+            return "NoJobType_%s_%s" % (siteName, jobType)
+
+        if highestPriorityInJobs is None or jobPrio <= highestPriorityInJobs:
+            # there is no pending or running jobs in the system (None case) or
+            # priority of the job is lower or equal don't allow overflow
+            totalPendingThreshold = totalPendingSlots
+            taskPendingThreshold = taskPendingSlots
+            totalJobThreshold = totalPendingSlots + totalRunningSlots
+            totalTaskTheshold = taskPendingSlots + taskRunningSlots
+        else:
+            # In case priority of the job is higher than any of currently pending or running.jobs
+            # increase the threshold by condorOverflowFraction * origianl pending slot.
+            pendingTotalOverflow = max(totalPendingSlots, initTotalPending) + (
+                                    totalPendingSlots * self.condorOverflowFraction)
+            pendingTaskOverflow = max(taskPendingSlots, initTaskPending) + (
+                                    taskPendingSlots * self.condorOverflowFraction)
+            totalPendingThreshold = pendingTotalOverflow
+            taskPendingThreshold = pendingTaskOverflow
+            totalJobThreshold = pendingTotalOverflow + totalRunningSlots
+            totalTaskTheshold = pendingTaskOverflow + taskRunningSlots
+
+        jobStats = [{"Condition": "NoPendingSlot",
+                     "Current": totalPendingJobs,
+                     "Threshold": totalPendingThreshold},
+                    {"Condition": "NoTaskPendingSlot",
+                     "Current": taskPendingJobs,
+                     "Threshold": taskPendingThreshold},
+                    {"Condition": "NoRunningSlot",
+                     "Current": totalPendingJobs + totalRunningJobs,
+                     "Threshold": totalJobThreshold},
+                    {"Condition": "NoTaskSlot",
```

Comment:
Just to keep it consistent, how about renaming it from `NoTaskSlot` to `NoTaskRunningSlot`?